Sources/FluentKit/Query/Builder/QueryBuilder+Join.swift

```diff
@@ -12,6 +12,72 @@ extension QueryBuilder {
         self.join(Foreign.self, filter.foreign, to: Local.self, filter.local , method: method)
     }
 
+    /// This will join a foreign table based on a `ParentProperty`relation
+    ///
+    /// This will not decode the joined data, but can be used in order to filter.
+    ///
+    ///     try Planet.query(on: db)
+    ///         .join(parent: \Planet.$star)
+    ///         .filter(Star.self, \Star.$name == "Sun")
+    ///
+    /// - Parameters:
+    ///   - parent: The `ParentProperty` to join
+    ///   - method: The method to use. The default is an inner join
+    /// - Returns: A new `QueryBuilder`
+    @discardableResult
+    public func join<From, To>(
+        parent: KeyPath<From, ParentProperty<From, To>>,
+        method: DatabaseQuery.Join.Method = .inner
+    ) -> Self {
+        join(To.self, on: parent.appending(path: \.$id) == \To._$id, method: method)
+    }
+
+    /// This will join a foreign table based on a `ChildrenProperty`relation
+    ///
+    /// This will not decode the joined data, but can be used in order to filter.
+    ///
+    ///     try Star.query(on: db)
+    ///         .join(children: \Planet.$planets)
```

Comment:
Shouldn't this be `\Star.$planets`?

Reply:
This also makes me wonder if we should have a set of more specific overloads for `QueryBuilder.Model` so that you can do:
```swift
Star.query(on: db).join(children: \.$planets)
```